pandas/core/groupby/generic.py

```diff
@@ -1056,7 +1057,8 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
     def _iterate_slices(self) -> Iterable[Series]:
         obj = self._selected_obj
         if self.axis == 1:
-            obj = obj.T
+            transposed_dtype = find_common_type(obj.dtypes.tolist())
+            obj = obj.T.astype(transposed_dtype, copy=False)
```

Comment:
why doesn't it already have this dtype?

Reply:
+1 on improving `.transpose` so that something like this is internal to it, but would be outside the scope of fixing this regression. I've opened #43337